client/sandbox/react-nextjs/pages/patterns/manage-users.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from "react";
+
+import { init, tx, id } from "@instantdb/react";
+import config from "../../config";
+import Login from "../../components/Login";
+
+const { auth, useAuth, transact, useQuery } = init(config);
+
+function App() {
+  const { isLoading, user, error } = useAuth();
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+  if (error) {
+    return <div>Uh oh! {error.message}</div>;
+  }
+  if (user) {
+    return <Main user={user} />;
+  }
+  return <Login auth={auth} />;
+}
+
+function Main({ user }: { user: { id: string; email: string } }) {
+  const { isLoading, error, data } = useQuery({ users: {} });
+  useEffect(() => {
+    if (isLoading) return;
+
+    // Don't add the user if it already exists
+    if (data?.users.find((u: any) => u.email === user.email)) return;
```

Comment:
I would remove this check, because it would require fetching all users. 

An alternative would be to query by the id, but in practice I don't think it should be necessary